lib/ecto/changeset.ex

```diff
@@ -1570,4 +1570,61 @@ defmodule Ecto.Changeset do
     model.__schema__(:association, assoc) ||
       raise(ArgumentError, "cannot add constraint to model because association `#{assoc}` does not exist")
   end
+
+  @doc """
+  Traverses changeset errors and applies function to error messages.
+
```

Comment:
Can we add a paragraph with something like this:

> This function is particularly useful when associations and embeds are cast
> in the changeset as it will traverse all associations and embeds and place
> all errors in a series of nested maps.